src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/AcceptanceImmigrantTranslator.java

```diff
@@ -0,0 +1,87 @@
+import uk.nhs.digital.nhsconnect.nhais.mapper.AcceptanceTypeMapper;
+import uk.nhs.digital.nhsconnect.nhais.mapper.GpNameAndAddressMapper;
+import uk.nhs.digital.nhsconnect.nhais.mapper.PartyQualifierMapper;
+import uk.nhs.digital.nhsconnect.nhais.mapper.PersonAddressMapper;
+import uk.nhs.digital.nhsconnect.nhais.mapper.PersonDateOfBirthMapper;
+import uk.nhs.digital.nhsconnect.nhais.mapper.PersonDateOfEntryMapper;
+import uk.nhs.digital.nhsconnect.nhais.mapper.PersonNameMapper;
+import uk.nhs.digital.nhsconnect.nhais.mapper.PersonSexMapper;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.BeginningOfMessage;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.DateTimePeriod;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.ReferenceTransactionNumber;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.ReferenceTransactionType;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.SegmentGroup;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static uk.nhs.digital.nhsconnect.nhais.mapper.FromFhirToEdifactMapper.emptyMapper;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class AcceptanceImmigrantTranslator implements FhirToEdifactTranslator {
+    private final PartyQualifierMapper partyQualifierMapper;
+    private final AcceptanceDateMapper acceptanceDateMapper;
+    private final GpNameAndAddressMapper gpNameAndAddressMapper;
+    private final AcceptanceCodeMapper acceptanceCodeMapper;
+    private final AcceptanceTypeMapper acceptanceTypeMapper;
+    private final PersonDateOfEntryMapper personDateOfEntryMapper;
+    private final PersonNameMapper personNameMapper;
+    private final PersonDateOfBirthMapper personDateOfBirthMapper;
+    private final PersonSexMapper personSexMapper;
+    private final PersonAddressMapper personAddressMapper;
+
+    @Override
+    public List<Segment> translate(Parameters parameters) throws FhirValidationException {
+        return Stream.of(
+            //BGM+++507'
+            emptyMapper(new BeginningOfMessage()),
+            //NAD+FHS+XX1:954'
+            partyQualifierMapper,
+            //DTM+137:199201151723:203'
+            acceptanceDateMapper,
+            //RFF+950:G1'
+            emptyMapper(new ReferenceTransactionType(ReferenceTransactionType.TransactionType.ACCEPTANCE)),
+            //S01+1'
+            emptyMapper(new SegmentGroup(1)),
+            //RFF+TN:20'
+            emptyMapper(new ReferenceTransactionNumber()),
+            //NAD+GP+2750922,295:900'
+            gpNameAndAddressMapper,
+            //HEA+ACD+A:ZZZ'
+            acceptanceCodeMapper,
+            //HEA+ATP+4:ZZZ'
+            acceptanceTypeMapper,
+            //DTM+956:19920115:102' (acceptance date ?)
+            emptyMapper(new DateTimePeriod(DateTimePeriod.TypeAndFormat.TRANSLATION_TIMESTAMP)),
```

Comment:
and this should be the `acceptanceDateMapper`. Swap this with the one above I've added comments on